examples/.wip/next-prisma-starter-websockets/src/pages/_app.tsx

```diff
@@ -14,34 +16,18 @@ const MyApp: AppType = ({ Component, pageProps }) => {
   );
 };
 
-function getHostname() {
-  if (process.browser) {
-    return 'localhost';
-  }
-  // // reference for vercel.com
-  // if (process.env.VERCEL_URL) {
-  //   return `${process.env.VERCEL_URL}`;
-  // }
-
-  // // reference for render.com
-  // if (process.env.RENDER_INTERNAL_HOSTNAME) {
-  //   return `${process.env.RENDER_INTERNAL_HOSTNAME}:${process.env.PORT}`;
-  // }
-
-  // assume localhost
-  return `http://localhost:${process.env.PORT ?? 3000}`;
-}
 function getEndingLink() {
-  if (process.browser) {
-    const client = createWSClient({
-      url: `ws://localhost:3001/api/trpc`,
-    });
-    return wsLink<AppRouter>({
-      client,
+  const { APP_URL } = publicRuntimeConfig;
+  if (!process.browser) {
+    return httpBatchLink({
+      url: `${APP_URL}/api/trpc`,
     });
   }
-  return httpBatchLink({
-    url: `${getHostname()}/api/trpc`,
+  const client = createWSClient({
+    url: APP_URL.replace(/^http/, 'ws'),
```

Comment:
This works, but I'd still be surprised to find out that the protocol got replaced.
I'd consider using a separate base URL for ws or exclude the protocol, even if those also have some drawbacks.